src/main/java/monday/task/Task.java

```diff
@@ -0,0 +1,57 @@
+package monday.task;
+
+import java.io.Serializable;
+
+/**
+ * The Task class represents a task with a description and a completion status.
+ */
+public class Task implements Serializable {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task object with the given description
+     * and sets its completion status to false.
+     *
+     * @param description the description of the task
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status icon of the task based on its completion status.
+     *
+     * @return "X" if the task is done, " " (empty space) otherwise
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    /**
+     * Marks the task as done by setting its completion status to true.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as not done by setting its completion status to false.
+     */
+    public void unMark() {
```

Comment:
I think there's an opportunity here to mirror the phrasings of related methods.
```suggestion
    public void markAsUndone() {
```

Reply:
That's a good suggestion, was feeling that the naming was unnatural too. Thank you!